src/utils/v2/math.ts

```diff
@@ -62,6 +62,17 @@ export const formatSplitPercent = (splitPercent: BigNumber): string => {
   ).toString()
 }
 
+/**
+ * Express a given split "percent" (parts-per-billion) as a percentage to the maximum precision.
+ * @param splitPercent - split "percent" as parts-per-billion.
+ * @returns {number} percentage (/100)
+ */
+export const preciseFormatSplitPercent = (
```

Comment:
Previous `formatSplitPercent` only precise to 2dp, which fucks up when deriving amounts. Made one that goes all the way

Reply:
Not sure if this was introduced in your code, but I am seeing a crash from line 85 of this file. It appears that a decimal is getting passed into `BigNumber.from` causing a crash. Recommend doing this:
```ts
BigNumber.from(Math.round(percentage * SPLITS_TOTAL_PERCENT / 100))
```

Reply:
Ahh yeah my bad I'll just revert that change actually I don't think it ended up being necessary